### PR TITLE
Balder/simplify rpc service pool

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli or @jonmv PR

### CODE REVIEW

**Review by jonmv — APPROVED** [review]
LGTM!

File: messagebus/src/main/java/com/yahoo/messagebus/network/rpc/RPCServicePool.java [hunk 1]
```diff
@@ -34,22 +37,28 @@ public RPCServicePool(RPCNetwork net, int maxSize) {
...
-        RPCService service = services.get().get(pattern);
+
+        ServiceLRUCache cache = getPerThreadCache();
+        RPCService service = cache.get(pattern);
         if (service == null) {
-            service = new RPCService(net.getMirror(), pattern);
-            services.get().put(pattern, service);
+            service = RPCService.create(net.getMirror(), pattern);
+            cache.put(pattern, service);
         }
```
Comment (jonmv) [inline messagebus/src/main/java/com/yahoo/messagebus/network/rpc/RPCServicePool.java:46]:
```suggestion
        RPRService service = cache.computeIfAbsent(pattern, key -> RPCService.create(net.getMirror(), key));
```

Comment (jonmv) [reply messagebus/src/main/java/com/yahoo/messagebus/network/rpc/RPCServicePool.java:46]: Huh, that diff makes no sense :)

Comment (baldersheim) [reply messagebus/src/main/java/com/yahoo/messagebus/network/rpc/RPCServicePool.java]: Simplify even more

**Review by jonmv — APPROVED** [review]

**Review by vekterli — APPROVED** [review]
:+1:

**Pull request merged**